src/mrpro/operators/LinearOperatorMatrix.py

```diff
@@ -0,0 +1,354 @@
+            raise ValueError('Other should have the same length as the operator has rows.')
+        else:
+            other_ = other
+        operators = []
+        for row, o in zip(self._operators, other_, strict=True):
+            operators.append([cast(LinearOperator, o * op) for op in row])
+        return self.__class__(operators)
+
+    def __matmul__(self, other: LinearOperator | Self) -> Self:  # type: ignore[override]
+        """Composition of operators."""
+        if isinstance(other, LinearOperator):
+            return self._binary_operation(other, '__matmul__')
+        elif isinstance(other, LinearOperatorMatrix):
+            if self.shape[1] != other.shape[0]:
+                raise ValueError('OperatorMatrix shapes do not match.')
+            new_operators = []
+            for row in self._operators:
+                new_row = []
+                for other_col in zip(*other._operators, strict=True):
+                    elements = [s @ o for s, o in zip(row, other_col, strict=True)]
+                    new_row.append(LinearOperatorSum(*elements))
+                new_operators.append(new_row)
+            return self.__class__(new_operators)
+        return NotImplemented  # type: ignore[unreachable]
+
+    @property
+    def H(self) -> Self:  # noqa N802
+        """Adjoints of the operators."""
+        return self.__class__([[op.H for op in row] for row in zip(*self._operators, strict=True)])
+
+    def adjoint(self, *x: torch.Tensor) -> tuple[torch.Tensor, ...]:
+        """Apply the adjoint of the operator to the input.
+
+        Parameters
+        ----------
+        x
+            Input tensors. Requires the same number of tensors as the operator has rows.
+
+        Returns
+        -------
+            Output tensors. The same number of tensors as the operator has columns.
+        """
+        return self.H(*x)
+
+    @classmethod
+    def from_diagonal(cls, *operators: LinearOperator):
+        """Create a diagonal LinearOperatorMatrix.
+
+        Create a square LinearOperatorMatrix with the given Linear Operators on the diagonal.
+
+        Parameters
+        ----------
+        operators
+            Sequence of Linear Operators to be placed on the diagonal.
+        """
+        operator_matrix: list[list[LinearOperator]] = [
+            [op if i == j else ZeroOp(False) for j in range(len(operators))] for i, op in enumerate(operators)
+        ]
+        return cls(operator_matrix)
+
+    def operator_norm(
+        self,
+        *initial_value: torch.Tensor,
+        dim: Sequence[int] | None = None,
+        max_iterations: int = 20,
+        relative_tolerance: float = 1e-4,
+        absolute_tolerance: float = 1e-5,
+        callback: Callable[[torch.Tensor], None] | None = None,
+    ) -> torch.Tensor:
+        """Upper bound of operator norm of the Matrix.
+
+        Uses the bounds :math:`||[A, B}^T|||<=sqrt(||A||^2 + ||B||^2)` and :math:`||[A, B]|||<=||A||+||B||`
```

Comment:
typo: [A, B]^T

Reply:
I think I do not really in which context these inequalities are used to compute the upper bound of the operator-norm.
Could we maybe give an example for an operator of the form, say, 
```
Op = [A,B,C]
     [D,E,F]
```
?

Reply:
added more exlantion and fixed typo